src/SSLSocket.c

```diff
@@ -541,6 +551,188 @@ static unsigned int call_ssl_psk_cb(SSL *ssl, const char *hint, char *identity,
 	return rc;
 }
 
+#ifdef PKCS11_HSM
+int pkcs11_hsm_load_pkey_from_hsm(SSL_CTX *ssl_ctx, MQTTClient_SSLOptions *sslopts)
+{
+    PKCS11_CTX *pkcs11_ctx;
+	PKCS11_SLOT *slots, *slot;
+    PKCS11_KEY *keys;
+	unsigned int nslots, nkeys;
+    EVP_PKEY *found_pkey = NULL;
+    const char *tokenlabel, *keylabel, *pinvalue;
+    int rc = 0;
+
+
+	pkcs11_ctx = sslopts->pkcs11_ctx;
+    slots = sslopts->pkcs11_slots;
+    nslots = sslopts->pkcs11_slot_num;
+    tokenlabel = sslopts->tokenLabel;
+    keylabel = sslopts->keyLabel;
+    pinvalue = sslopts->pinValue;
+
+	slot = PKCS11_find_token(pkcs11_ctx, slots, nslots);
+    for (unsigned int i = 1; i <= nslots; i++) {
+        if (slot == NULL || slot->token == NULL) {
+            fprintf(stderr, "no token!\n");
```

Comment:
Why is there output to stderr rather than the log?  Is this a utility/test or similar?

Reply:
Yeah this was left over from some testing we had done (based on code from the p11 project).  I've removed the `printf`s